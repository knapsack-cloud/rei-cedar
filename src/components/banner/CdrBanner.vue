<template>
  <div :class="[$style[baseClass], $style[typeClass]]">
    <slot />
  </div>
</template>

<script>
import { defineComponent, computed } from 'vue';
import propValidator from '../../utils/propValidator';
import { buildClass } from '../../utils/buildClass';

export default defineComponent({
  name: 'CdrBanner',
  props: {
    type: {
      type: String,
      validator: (value) => propValidator(
        value,
        ['info', 'warning', 'success', 'error'],
      ),
      default: 'info',
    },
  },
  setup(props) {
    const baseClass = 'cdr-banner';
    const typeClass = computed(() => props.type && buildClass(baseClass, props.type));
    return {
      baseClass,
      typeClass,
    };
  },
});
</script>

<style lang="scss" module src="./styles/CdrBanner.module.scss">
</style>
