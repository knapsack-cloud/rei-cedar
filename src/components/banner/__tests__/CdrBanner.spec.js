import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrBanner from 'componentdir/banner/CdrBanner';
// import IconCheckFill from 'componentdir/icon/comps/check-fill';
describe('CdrBanner', () => {
  it('matches snapshot', () => {
    const wrapper = mount(CdrBanner, {
      propsData: {
        type: 'info',
      },
      slots: {
       default: "ðŸ¤ª hey im a banner",
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
