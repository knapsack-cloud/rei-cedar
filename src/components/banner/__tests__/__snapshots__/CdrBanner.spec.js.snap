// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrBanner matches snapshot 1`] = `
<div
  class="cdr-banner cdr-banner--info"
>
  
  ðŸ¤ª hey im a banner
  
</div>
`;
